tests/arp/test_neighbor_mac.py

```diff
@@ -0,0 +1,181 @@
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        logger.info("Configure the DUT interface, start interface, add IP address")
+        self.__startInterface(duthost)
+        self.__configureInterfaceIp(duthost, action="add")
+
+        yield
+
+        logger.info("Restore the DUT interface config, remove IP address")
+        self.__configureInterfaceIp(duthost, action="remove")
+
+    @pytest.fixture(params=[0, 1])
+    def macIndex(self, request):
+        """
+            Parameterized fixture for macIndex
+
+            Args:
+                request: pytest request object
+
+            Retruns:
+                macIndex (int): index of the mac address used from TEST_MAC
+        """
+        yield request.param
+
+    def __configureNeighborIp(self, ptfhost, macIndex):
+        """
+            Configure interface and set IP address/mac address on the PTF host
+
+            Args:
+                ptfhost (PTF host): PTF instance used
+                macIndex (int): test MAC index to be used
+
+            Returns:
+                None
+        """
+        ptfhost.shell("ifconfig {} {} netmask 255.255.255.0".format(self.PTF_HOST_IF, self.PTF_HOST_IP))
+        neighborMac = self.TEST_MAC[macIndex]
+        logger.info("neighbor {0} lladdr {1} for {2}".format(self.PTF_HOST_IP, neighborMac, self.PTF_HOST_IF))
+        ptfhost.shell("ifconfig {} down".format(self.PTF_HOST_IF))
+        ptfhost.shell("ifconfig {} hw ether {}".format(self.PTF_HOST_IF, neighborMac))
+        ptfhost.shell("ifconfig {} up".format(self.PTF_HOST_IF))
+
+    def __startInterface(self, duthost):
+        """
+            Startup the interface on the DUT
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+
+            Returns:
+                None
+        """
+        logger.info("Configure the interface '{0}' as UP".format(self.DUT_ETH_IF))
+        duthost.shell(argv=[
+            "config",
+            "interface",
+            "startup",
+            self.DUT_ETH_IF
+        ])
+
+    def __configureInterfaceIp(self, duthost, action=None):
+        """
+            Configure interface IP address on the DUT
+
+            Args:
+                duthost (AnsibleHost): Device Under Test (DUT)
+                action (str): action to perform, add/remove interface IP
+
+            Returns:
+                None
+        """
+
+        logger.info("{0} an ip entry {1} for {2}".format(action, self.DUT_INTF_IP, self.DUT_ETH_IF))
+        interfaceIp = "{}/{}".format(self.DUT_INTF_IP, self.DUT_INTF_NETMASK)
+        duthost.shell(argv=[
+            "config",
+            "interface",
+            "ip",
+            action,
+            self.DUT_ETH_IF,
+            interfaceIp
+        ])
+
+    @pytest.fixture(autouse=True)
+    def configureNeighborIpAndPing(self, duthost, ptfhost, macIndex):
```

Comment:
Are we using camel case not only in Class naming but also in function arguments naming  (macIndex -> mac_index)?